### PR TITLE
Implement expires_at in Domain struct

### PR DESCRIPTION
**Pull request by duduribeiro**
We deprecated expires_on attribute in our api and in docs in favor of
expires_at. This commit introduces the new expires_at attribute.

Also updated the fixtures with the new field and removing the old
premium_price from Domain Transfer/Register/Renewal.

### CODE REVIEW

File: dnsimple/domains.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ type Domain struct {
 	AutoRenew    bool   `json:"auto_renew,omitempty"`
 	PrivateWhois bool   `json:"private_whois,omitempty"`
 	ExpiresOn    string `json:"expires_on,omitempty"`
```
Comment (duduribeiro) [inline dnsimple/domains.go:27]: I wanted to make a deprecation warn in this attribute but couldn't find a good way on do it 🤔. any ideas?

**Comment by duduribeiro** [comment]
We are trying to implement the same thing we implemented for the ruby one: https://github.com/dnsimple/dnsimple-ruby/pull/186 (stoping using the expiresOn attribute from the json response, and makes the expiresOn to be set based on the expiresAt value)

I still don't know if this is the best way to set expiresOn attribute in the struct. 

What I did:
- Stop it from using the json value: https://github.com/dnsimple/dnsimple-go/pull/98/files#diff-8074276759f70ae6493171bb178505e5R27

- Created the method to set expiresOn based on expiresAt https://github.com/dnsimple/dnsimple-go/pull/98/files#diff-8074276759f70ae6493171bb178505e5R124

- Changed the callers to execute this method after retrieving the http response. For the List one, I needed to iterate from the result to execute the method https://github.com/dnsimple/dnsimple-go/pull/98/files#diff-8074276759f70ae6493171bb178505e5R82. IDK if it is the best option.

**Comment by duduribeiro** [comment]
Followed with @onlyhavecans proposed. Leave expires_on parsing the field and leave the deprecation note.

File: dnsimple/domains_test.go [hunk 2]
```diff
@@ -48,12 +48,16 @@ func TestDomainsService_ListDomains(t *testing.T) {
...
 
-	if want, got := int64(1), domains[0].ID; want != got {
+	if want, got := int64(181984), domains[0].ID; want != got {
 		t.Fatalf("Domains.ListDomains() returned ID expected to be `%v`, got `%v`", want, got)
 	}
 	if want, got := "example-alpha.com", domains[0].Name; want != got {
 		t.Fatalf("Domains.ListDomains() returned Name expected to be `%v`, got `%v`", want, got)
 	}
+
+	if want, got := "2021-06-05", domains[0].ExpiresOn; want != got {
```
Comment (weppos) [inline dnsimple/domains_test.go:58]: @onlyhavecans @DXTimer @duduribeiro I see two different attributes referenced in the same spec (aka the message doesn't match the code). Is that on purpose?

**Comment by weppos** [comment]
> I believe the client should reflect the behavior of the API, not hot-patch the API's behavior.

@onlyhavecans this is unfortunately not always possible, unless we provision a new version of the API on every change. Clients are the only way we have right now to reliably enforce/expose deprecations.

**Comment by weppos** [comment]
The Go client is still not in 1.x, so we can still make some breaking change. Given Go is actually more strict than other languages and don't easily allow to replace an attribute with a virtual setter/getter, I would actually be tempted to simply drop expires_on entirely and replace it with expires_at.

We will have to clearly document it in the README, and release 0.70.x as a sign of change.

File: dnsimple/domains_test.go (hunk 2 above)
Comment (duduribeiro) [reply dnsimple/domains_test.go:58]: No, my fault!. should be expiresOn.

**Comment by duduribeiro** [comment]
@weppos I dropped the ExpiresOn attribute like you suggested. Also updated the CHANGELOG

**Pull request merged**